### PR TITLE
add: Cloud DNS SSL config

### PR DESCRIPTION
**Pull request by ericpaulsen**
adding steps for configuring SSL for Coder via Google's Cloud DNS, referenced from [this doc](https://docs.google.com/document/d/1NahF4Mx5t_nLrwg7MZsyeSlMVnEIQi29v7poNl6LV8w/edit#).

i've completed a successful install for a customer using the linked doc, but it wouldn't hurt for someone else to dogfood it.

please see my comments below for line-specific thoughts.

### CODE REVIEW

File: guides/ssl-certificates/cloudDNS.md [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+you can enable HTTPS on your Coder deployment. It will also show you how to
+configure your Coder hostname and dev URLs.
+
+> We recommend reviewing the official cert-manager
+> [documentation](https://cert-manager.io/docs/) if you encounter any issues or
+> if you want info on using a different certificate issuer.
+
+You must have:
+
+- A Kubernetes cluster with internet connectivity
```
Comment (ericpaulsen) [inline guides/ssl-certificates/cloudDNS.md:23]: `cert-manager` supports 1.16+ K8s versions, but I could not find any Cloud DNS-specific version compatibilities. This doc also assumes that a customer has a Service Account configured.

File: guides/ssl-certificates/cloudDNS.md [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+- [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
+- A
+  [Service Account](https://cloud.google.com/iam/docs/creating-managing-service-accounts)
+
+## Step 1: Add cert-manager to your Kubernetes cluster
```
Comment (ericpaulsen) [inline guides/ssl-certificates/cloudDNS.md:28]: If needed, feel free to add context around applying the `cert-manager` manifest.

File: guides/ssl-certificates/route53.md [hunk 3]
```diff
@@ -142,7 +142,7 @@ helm install coder coder/coder --namespace coder \
   --set devurls.host="*.exampleCo.com" \
   --set ingress.host="coder.exampleCo.com" \
   --set ingress.tls.enable=true \
-  --set ingress.tls.devUrlsHostSecretName=devUlrCertificate \
+  --set ingress.tls.devUrlsHostSecretName=devUrlCertificate \
```
Comment (ericpaulsen) [inline guides/ssl-certificates/route53.md:145]: typo fix.

File: guides/ssl-certificates/cloudDNS.md [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+> We recommend reviewing the official cert-manager
+> [documentation](https://cert-manager.io/docs/) if you encounter any issues or
+> if you want info on using a different certificate issuer.
+
+You must have:
+
+- A Kubernetes cluster with internet connectivity
+- [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
+- A
+  [Service Account](https://cloud.google.com/iam/docs/creating-managing-service-accounts)
```
Comment (bpmct) [inline guides/ssl-certificates/cloudDNS.md:26]:
```suggestion
  [GCP Service Account](https://cloud.google.com/iam/docs/creating-managing-service-accounts) with the `dns.admin` role
```

**Review by bpmct — CHANGES_REQUESTED** [review]
This is awesome Eric!

I think this also needs to be added to the manifest.json so that it is added in https://codercom-2h12dwib3-codercom.vercel.app/docs/guides/ssl-certificates

File: guides/ssl-certificates/cloudDNS.md (hunk 4 above)
Comment (bpmct) [reply guides/ssl-certificates/cloudDNS.md:26]: context: https://cert-manager.io/docs/configuration/acme/dns01/google/#set-up-a-service-account

**Review by khorne3 — APPROVED** [review]

**Pull request merged**